pyp2rpm/filters.py

```diff
@@ -62,10 +62,19 @@ def package_to_path(package, module):
         return package
 
 
+def macroed_url(url):
+    if (url.startswith('https://files.pythonhosted.org/packages/source/') and
+        (url.endswith('/%{pypi_name}/%{pypi_name}-%{version}.tar.gz') or
+         url.endswith('/%{pypi_name}/%{pypi_name}-%{version}.zip'))):
```

Comment:
If it ends with zip, you should return `%{pypi_source %{pypi_name} %{version} zip}` instead.